cli/cli_app.go

```diff
@@ -223,6 +224,41 @@ func runTerragrunt(terragruntOptions *options.TerragruntOptions) error {
 	return runTerragruntWithConfig(terragruntOptions, terragruntConfig, false)
 }
 
+func processHooks(hooks []config.Hook, terragruntConfig *config.TerragruntConfig, terragruntOptions *options.TerragruntOptions, previousExecError ...error) error {
+
+	terraformCommand := terragruntOptions.TerraformCliArgs[0]
+
+	errorsOccurred := []error{}
+
+	if len(hooks) > 0 {
+		terragruntOptions.Logger.Printf("Detected %d Hooks", len(hooks))
+
+		for i := 0; i < len(hooks); i++ {
+			applicableCommands := hooks[i].Commands
+
+			if util.ListContainsElement(applicableCommands, terraformCommand) {
+				//if there's no previous error, execute command
+				//OR if a previos error DID happen AND we want to run anyways
+				//then execute.
+				//Skip execution if there was an error AND we care about errors
+				if len(previousExecError) == 0 || hooks[i].RunOnError {
```

Comment:
To reduce nesting, could combine these if-statements into a method and call this method in the for-loop:

```go
func shouldRunHook(hook config.Hook, terragruntOptions *options.TerragruntOptions,  previousExecErrors ... error) bool {
  return util.ListContainsElement(hook.Commands, terragruntOptions..TerraformCliArgs[0]) && (len(previousExecErrors) == 0) || hook.RunOnError)
}
```